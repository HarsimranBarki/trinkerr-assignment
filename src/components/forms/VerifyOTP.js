import { motion } from "framer-motion";
import React, { useState } from "react";
import toast, { Toaster } from "react-hot-toast";

const VerifyOTP = ({ setSignupState }) => {
  const [otp, setOTP] = useState("");
  const toastGenerate = (message) => toast.error(message);

  // Regex For Formatting Input Should Be Number
  const onlyNumbers = (e) => {
    e.preventDefault();
    let value = e.target.value.replace(/[^\d]/g, "");
    setOTP(value);
  };

  const validateInput = (e) => {
    e.preventDefault();
    if (otp.length !== 4) return toastGenerate("OTP should be 4 Digits");
    if (otp !== "0000") return toastGenerate("Wrong OTP");
    setSignupState("name");
  };

  return (
    <motion.form
      className="mt-10 "
      onSubmit={(e) => validateInput(e)}
      initial={{ opacity: 0, x: -10 }}
      animate={{ opacity: 1, x: 0 }}
      exit={{ opacity: 0, x: 10 }}
    >
      <Toaster position="bottom-center" reverseOrder={false} />
      <input
        type="text"
        className="form-input px-4 py-2 border-gray-300 rounded mt-2 shadow-inner"
        placeholder="Enter Your OTP"
        value={otp}
        onChange={(e) => onlyNumbers(e)}
        autoFocus
      />
      <br />
      <button
        type="submit"
        className="bg-blue-600 text-indigo-50 px-5 py-2  font-medium rounded mt-5 hover:bg-blue-700 transition"
      >
        Submit OTP
      </button>
    </motion.form>
  );
};

export default VerifyOTP;
