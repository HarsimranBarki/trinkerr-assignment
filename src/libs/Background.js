const Background = ({ children }) => {
  return (
    // Remove transition-all to disable the background color transition.
    <div className="bg-white dark:bg-gray-800 dark:text-gray-100 h-screen transition-all">
      {children}
    </div>
  );
};
export default Background;
