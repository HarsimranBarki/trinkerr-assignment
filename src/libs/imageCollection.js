import one from "../images/one.png";
import two from "../images/two.png";
import three from "../images/three.png";
import four from "../images/four.png";
import five from "../images/five.png";

export const imageCollection = [
  {
    id: 1,
    name: "one",
    imageUrl: one,
    swiped: false,
  },
  {
    id: 2,
    name: "two",
    imageUrl: two,
    swiped: false,
  },
  {
    id: 3,
    name: "three",
    imageUrl: three,
    swiped: false,
  },
  {
    id: 4,
    name: "four",
    imageUrl: four,
    swiped: false,
  },
  {
    id: 5,
    name: "five",
    imageUrl: five,
    swiped: false,
  },
];
