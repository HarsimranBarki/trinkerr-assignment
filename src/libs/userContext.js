import React, { useMemo, useState } from "react";

export const UserContext = React.createContext(null);

export const UserProvider = ({ children }) => {
  const [user, setUser] = useState(
    JSON.parse(localStorage.getItem("user")) || null
  );
  const providerValue = useMemo(() => ({ user, setUser }), [user, setUser]);

  return (
    <UserContext.Provider value={providerValue}>
      {children}
    </UserContext.Provider>
  );
};
