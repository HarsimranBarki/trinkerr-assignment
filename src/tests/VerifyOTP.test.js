import { fireEvent, render, screen } from "@testing-library/react";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Enzyme, { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";
import VerifyOTP from "../components/forms/VerifyOTP";
import { UserProvider } from "../libs/userContext";

Enzyme.configure({ adapter: new Adapter() });

describe("VerifyOTP", () => {
  const setSignupState = jest.fn();
  it("VerifyOTP: should render correctly", () => {
    const wrapper = shallow(
      <UserProvider>
        <VerifyOTP setSignupState={setSignupState} />
      </UserProvider>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("VerifyOTP: should not be greater or less than 4", () => {
    render(
      <UserProvider>
        <VerifyOTP setSignupState={setSignupState} />
      </UserProvider>
    );

    const buttonPhone = screen.queryByTestId("button");
    const inputOTP = screen.queryByTestId("input");

    fireEvent.change(inputOTP, { target: { value: "0000" } });
    fireEvent.click(buttonPhone);

    expect(inputOTP.value).toBe("0000");

    let vals = inputOTP.value.length;
    expect(vals).not.toBeGreaterThan(4);
    expect(vals).not.toBeLessThan(4);
  });

  it("VerifyOTP: should not allow null values", () => {
    render(
      <UserProvider>
        <VerifyOTP setSignupState={setSignupState} />
      </UserProvider>
    );
    const buttonPhone = screen.queryByTestId("button");
    const inputOTP = screen.queryByTestId("input");

    inputOTP.value = "";
    fireEvent.click(buttonPhone);
    let vals = inputOTP.value.length;
    // buttonPhone.simulate("click");
    expect(vals).not.toBeNull();
  });
});
