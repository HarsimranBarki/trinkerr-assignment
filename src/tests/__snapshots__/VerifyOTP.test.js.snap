// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyOTP VerifyOTP: should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "setUser": [Function],
      "user": null,
    }
  }
>
  <VerifyOTP
    setSignupState={[MockFunction]}
  />
</ContextProvider>
`;
